services/provider/server/consumer.go

```diff
@@ -55,6 +58,43 @@ func newConsumerManager(ctx context.Context, cl client.Client, namespace string)
 	}, nil
 }
 
+// getStorageRequestHash generates a hash for a StorageRequest based
+// on the MD5 hash of the StorageClaim name and storageConsumer UUID.
+func getStorageRequestHash(consumerUUID, storageClaimName string) string {
+	s := struct {
+		StorageConsumerUUID string `json:"storageConsumerUUID"`
+		StorageClaimName    string `json:"storageClaimName"`
+	}{
+		consumerUUID,
+		storageClaimName,
+	}
+
+	requestName, err := json.Marshal(s)
+	if err != nil {
+		klog.Errorf("failed to marshal a name for a storage class request based on %v. %v", s, err)
+		panic("failed to marshal storage class request name")
+	}
+	md5Sum := md5.Sum(requestName)
+	return hex.EncodeToString(md5Sum[:16])
+}
+
+// getStorageRequestName generates a name for a StorageRequest resource.
+func getStorageRequestName(consumerUUID, storageClaimName string) string {
+	return fmt.Sprintf("storagerequest-%s", getStorageRequestHash(consumerUUID, storageClaimName))
+}
```

Comment:
Do we really need these 2 functions as standalone functions? I see the logic belong for a single flow which is the `GetRequest` logic. Can we fold everything there? 
It will make it easier to move around and potentially remove in the future.